pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -1026,6 +1033,64 @@ func PodEnv(i *infinispanv1.Infinispan, systemEnv *[]corev1.EnvVar) []corev1.Env
 	return envVars
 }
 
+func applyExternalArtifactsDownload(ispn *infinispanv1.Infinispan, spec *corev1.PodSpec) (updated bool) {
+	c := &spec.InitContainers
+	volumes := &spec.Volumes
+	volumeMounts := &spec.Containers[0].VolumeMounts
+	if ispn.HasExternalArtifacts() {
+		extractCommands := externalArtifactsExtractCommand(ispn, ExternalArtifactsMountPath)
+		if position := findInitContainer(*c, ExternalArtifactsDownloadInitContainer); position >= 0 {
+			if spec.InitContainers[position].Args[0] != extractCommands {
+				spec.InitContainers[position].Args = []string{extractCommands}
+				updated = true
+			}
+		} else {
+			*c = append(*c, corev1.Container{
+				Image:   ispn.ImageName(),
+				Name:    ExternalArtifactsDownloadInitContainer,
+				Command: []string{"sh", "-c"},
+				Args:    []string{extractCommands},
+				VolumeMounts: []corev1.VolumeMount{{
+					Name:      ExternalArtifactsVolumeName,
+					MountPath: ExternalArtifactsMountPath,
+				}},
+			})
+			*volumeMounts = append(*volumeMounts, corev1.VolumeMount{Name: ExternalArtifactsVolumeName, MountPath: ExternalArtifactsMountPath, ReadOnly: true})
+			*volumes = append(*volumes, corev1.Volume{Name: ExternalArtifactsVolumeName, VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}})
+			updated = true
+		}
+	} else {
+		if position := findInitContainer(*c, ExternalArtifactsDownloadInitContainer); position >= 0 {
+			volumePosition := findVolume(*volumes, ExternalArtifactsVolumeName)
+			volumeMountPosition := findVolumeMount(*volumeMounts, ExternalArtifactsVolumeName)
+			*c = append(spec.InitContainers[:position], spec.InitContainers[position+1:]...)
+			*volumes = append(spec.Volumes[:volumePosition], spec.Volumes[volumePosition+1:]...)
+			*volumeMounts = append(spec.Containers[0].VolumeMounts[:volumeMountPosition], spec.Containers[0].VolumeMounts[volumeMountPosition+1:]...)
+			updated = true
+		}
+	}
+	return
+}
+
+func externalArtifactsExtractCommand(ispn *infinispanv1.Infinispan, mountPath string) string {
```

Comment:
No need for `mountPath` to be a parameter as it's always `ExternalArtifactsMountPath`.